File: greeting/send.js
const amqp = require('amqplib');

/**
 * This is an immediately invoked function expression to be used as async
 * function because of we need to use 'await' keyword
 */
(async () => {
    let connection = await amqp.connect('amqp://localhost');
    let channel = await connection.createChannel();
    let queue = 'greeting';
    let message = 'Hi, there!';

    await channel.assertQueue(queue, {
        durable: false
    });

    channel.sendToQueue(queue, Buffer.from(message));
    console.log(` [x] Sent ${message}`);

    setTimeout(async () => {
        await connection.close();
        process.exit(0);
    }, 500);
})();

File: greeting_python/send.py
import pika

connection = pika.BlockingConnection(pika.ConnectionParameters('localhost'))
channel = connection.channel()
queue = 'greeting'
message = 'Hi, there!'

channel.queue_declare(queue=queue)
channel.basic_publish(exchange='', routing_key=queue, body=message)
print(' [X] Sent', message)
connection.close()
